integration-tests/tests/src/tests/mixed.ts

```diff
@@ -0,0 +1,312 @@
+      expect(data.c).equals(true, "should store boolean (true)");
+    });
+
+    it("support queries on mixed primitives", function (this: RealmContext) {
+      const N = 100;
+      for (let i = 0; i < N; i++) {
+        this.realm.write(() => this.realm.create(SingleSchema.name, { a: "xxxxxx", b: "xxx", c: i }));
+      }
+
+      const data = this.realm.objects(SingleSchema.name);
+      const section = data.filtered('a BEGINSWITH "x" AND b BEGINSWITH "x"');
+      const half = data.filtered("c < 50");
+      expect(section.length).equals(N, "We expect only 100 items.");
+      expect(half.length).equals(N / 2, "We expect only 50 items.");
+    });
+
+    it("support complex primitive types", function (this: RealmContext) {
+      const d128 = Decimal128.fromString("6.022e23");
+      const oid = new ObjectId();
+      const uuid = new UUID();
+      const date = new Date();
+
+      const data: ISingleSchema = this.realm.write(() =>
+        this.realm.create(SingleSchema.name, { a: oid, b: uuid, c: d128, d: date }),
+      );
+
+      expect(typeof data.a === typeof oid, "should be the same type ObjectId");
+      expect(String(data.a)).equals(oid.toString(), "should be the same ObjectId");
+      expect(String(data.b)).equals(uuid.toString(), "Should be the same UUID");
+      expect(String(data.c)).equals(d128.toString(), "Should be the same Decimal128");
+      expect(String(data.d)).equals(date.toString(), "Should be the same Date");
+    });
+
+    it("support mixed mutability", function (this: RealmContext) {
+      const d128 = Decimal128.fromString("6.022e23");
+      const oid = new ObjectId();
+      const uuid = new UUID();
+      const date = new Date();
+
+      const data: ISingleSchema = this.realm.write(() =>
+        this.realm.create<ISingleSchema>(SingleSchema.name, { a: oid }),
+      );
+      expect(typeof data.a === typeof oid, "should be the same type ObjectId");
+      expect(String(data.a)).equals(oid.toString(), "should have the same content");
+
+      this.realm.write(() => (data.a = uuid));
+      expect(typeof data.a === typeof uuid, "should be the same type UUID");
+      expect(String(data.a)).equals(uuid.toString(), "should have the same content UUID");
+
+      this.realm.write(() => (data.a = d128));
+      expect(String(data.a)).equals(d128.toString(), "Should be the same Decimal128");
+
+      this.realm.write(() => (data.a = 12345678));
+      expect(data.a).equals(12345678, "Should be the same 12345678");
+
+      this.realm.write(() => ((data.a = null), "Should be the same null"));
+      expect(data.a).equals(null);
+
+      this.realm.write(() => ((data.a = undefined), "Should be the same null"));
+      expect(data.a).equals(null);
+    });
+    it("wrong type throws", function (this: RealmContext) {
+      expect(() => {
+        this.realm.write(() => this.realm.create(SingleSchema.name, { a: Object.create({}) }));
+      }).throws(Error, "Only Realm instances are supported.");
```

Comment:
in the previous branch the error message thrown was `Unable to convert an object with ctor 'Object' to a Mixed` which makes way more sense.. I don't know if I should either 1: leave it as is with an unclear error message, 2: change back to the regular string and leave the test as failing or 3: deal with why the wrong message is thrown.